src/App.tsx

```diff
@@ -1,27 +1,62 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import './reset.scss';
 import './App.scss';
+import ButtonGroup from '@mui/material/ButtonGroup';
+import Button from '@mui/material/Button';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRed } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const getGoods = useCallback(async (action = getAll) => {
```

Comment:
useCallback doesn't improve anything here, no effects or `memo` components depend on it, you pass an inline function (`onClick={() => getGoods()}`) to buttons anyway.